docs/source/guides/build-options.rst

```diff
@@ -170,6 +170,29 @@ BentoML will lock the package to the version available in the current environmen
     You don't need to specify ``bentoml`` as a dependency in this field since the current version of BentoML will be added to the list by default. However,
     you can override this by specifying a different BentoML version.
 
+To include a package from a GitHub repository, use the `pip VCS support format <https://pip.pypa.io/en/stable/topics/vcs-support/>`_. You can specify the repository URL, the branch, tag, or commit to install from, and the subdirectory if the Python package is not in the root of the repository.
```

Comment:
The formats are the same as pip's requirements file: https://pip.pypa.io/en/stable/reference/requirements-file-format/